addons/pos_gift_card/models/gift_card.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class GiftCard(models.Model):
+    _inherit = "gift.card"
+
+    buy_pos_order_line_id = fields.Many2one(
+        "pos.order.line",
+        copy=False,
+        readonly=True,
+        help="Pos Order line where this gift card has been bought.",
+    )
+    redeem_pos_order_line_ids = fields.One2many(
+        "pos.order.line", "gift_card_id", string="Pos Redeems"
+    )
+
+    @api.depends("redeem_pos_order_line_ids")
+    def _compute_balance(self):
+        super()._compute_balance()
+        for record in self:
+            confirmed_line = record.redeem_pos_order_line_ids.filtered(
+                lambda l: l.order_id.state == "paid"
+            )
+            balance = record.balance
+            if confirmed_line:
```

Comment:
how can int works in // with sale ? 

Shouln't with have the inital balance -sale -pos ? 
Here the last called won ... no?